sh/tests/file-checker_test.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS='
+'
+
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
```

Comment:
Here create the func Challenge for us to test multiple types of files and not just one.
```
challenge() {
    submitted=$(bash "$script_dirS"/student/file-checker.sh $1)
    expected=$(bash "$script_dirS"/solutions/file-checker.sh $1)
    diff <(echo "$submitted") <(echo "$expected")
}
```
Then create a new folder with the same name as the exercise and add multiple files inside it to test they are or not readable exec etc...

then you can run it like:
```
challenge file-checker/readable-only
...
```

Reply:
Thanks, done. I added all the combinations of permissions that git allowed me to. I found that I can only add a file to git if they are at least readable. Do you know if it's possible to add files that don't have the permission to read?

Reply:
not sure, but i think it should be possible...
Still the files you created have some wrong perms
![image](https://user-images.githubusercontent.com/53977751/211862030-8915d308-ac5c-4f33-8082-bf161ff09cd0.png)

Reply:
That is interesting: 
![image](https://user-images.githubusercontent.com/18433970/211866758-b6ef0ae7-b08c-48bf-9983-b62354b3d1b3.png)

Reply:
I ran this right now:
![image](https://user-images.githubusercontent.com/18433970/211867437-ba9098ad-5fb4-40c2-a045-1ab49dc24704.png)

Reply:
Maybe when pushing it the permissions change? or when pulling it?

Reply:
I tried deleting the files, and changing permissions before staging the changes to git.

Reply:
yes something is definitely happening here because when i pull the files have a wrong format

Reply:
the solution for this can be to create the files in the test i think

Reply:
Thanks. That was a good idea that triggered a better one I think. Instead what I am doing is setting the permissions right before running the tests, no need to create the files every time. Let me know if you like it, and if works on your end. It's working locally. Otherwise I'll change it. :+1: